src/controller/OperationalCredentialsDelegate.h

```diff
@@ -66,6 +66,34 @@ class DLL_EXPORT OperationalCredentialsDelegate
                                         const ByteSpan & DAC, const ByteSpan & PAI,
                                         Callback::Callback<OnNOCChainGeneration> * onCompletion) = 0;
 
+    /**
+     * @brief
+     *   This function generates an operational certificate chain for a remote device that is being commissioned.
+     *   The API generates the certificate in CHIP cert form.
+     *
+     *   The delegate is expected to use the certificate authority whose certificate
+     *   is returned in `GetRootCACertificate()` API call.
+     *
+     *   The delegate will call `onCompletion` when the NOC certificate chain is ready.
+     *
+     * @param[in] csrElements          CSR elements as per specifications section 11.18.5.6. NOCSR Elements.
+     * @param[in] csrNonce             CSR nonce as described in 6.4.6.1
+     * @param[in] attestationSignature Attestation signature as per specifications section 11.22.7.6. CSRResponse Command.
+     * @param[in] attestationChallenge Attestation challenge as per 11.18.5.7
+     * @param[in] DAC                  Device attestation certificate received from the device being commissioned
+     * @param[in] PAI                  Product Attestation Intermediate certificate
+     * @param[in] onCompletion         Callback handler to provide generated NOC chain to the caller of GenerateNOCChain()
+     *
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    virtual CHIP_ERROR GenerateChipNOCChain(const ByteSpan & csrElements, const ByteSpan & csrNonce,
```

Comment:
This is adding a footgun: now operational credentials delegates don't have to implement this function, but the core might end up using it (because in the end commissioning _does_ want the NOC chain in Matter TLV form).

We should either require all credentials delegates to support both, or have this API have only one callback and fix the code that is being added to do the DER to TLV conversions itself (or vice versa if we decide the only one to support the TLV form), or have the defult impl of the "non-canonical" NOC chain getter call the other getter and convert.

Reply:
Fixed in https://github.com/project-chip/connectedhomeip/pull/20013